plugins/woocommerce/tests/e2e-pw/tests/merchant/product-delete.spec.js

```diff
@@ -0,0 +1,121 @@
+const {test: baseTest, expect} = require('@playwright/test');
+const wcApi = require('@woocommerce/woocommerce-rest-api').default;
+
+baseTest.describe('Products > Delete Product', () => {
+	baseTest.use({storageState: process.env.ADMINSTATE});
+
+	const apiFixture = baseTest.extend({
+		api: async ({baseURL}, use) => {
+			const api = new wcApi({
+				url: baseURL,
+				consumerKey: process.env.CONSUMER_KEY,
+				consumerSecret: process.env.CONSUMER_SECRET,
+				version: 'wc/v3',
+				axiosConfig: {
+					// allow 404s, so we can check if the product was deleted without try/catch
+					validateStatus: function (status) {
+						return status >= 200 && status < 300 || status === 404;
+					},
+				},
+			});
+
+			await use(api);
+		},
+	});
+
+	const test = apiFixture.extend({
+		productId: async ({baseURL, api}, use) => {
+			let productId;
+
+			await api
+				.post('products', {
+					name: 'Product to delete',
+					type: 'simple',
+					regular_price: '12.99',
+				})
+				.then((response) => {
+					productId = response.data.id;
+				});
+
+			await use(productId);
+
+			// permanently delete the product if it still exists
+			const r = await api.get(`products/${productId}`);
+			if (r.status !== 404) {
+				await api.delete(`products/${productId}`, {
+					force: true,
+				});
+			}
+		},
+	});
+
+	test('can delete a product from edit view', async ({page, productId, api}) => {
+		await test.step('Navigate to product edit page', async () => {
+			await page.goto(`wp-admin/post.php?post=${productId}&action=edit`);
+		});
+
+		await test.step('Move product to trash', async () => {
+			await page.locator('#delete-action > a').click();
+		});
+
+		await test.step('Verify product was trashed', async () => {
+			await expect(
+				page.locator('#message').last()
```

Comment:
I don't think this is a safe bet for stability. There may be multiple messages shown.
Instead, what about using `getByText`:
`await expect(page.getByText( '1 product moved to the Trash.' )).toBeVisible();`
Please let me know if that works for you.

Reply:
I was actually trying to be explicit in the check of the message being the last one, hence the use of `.last()`. My idea was that it might be good for the test to fail if the message is not the last one, because it might indicate on issue worth investigating. 
Running tests in this file in parallel could possibly impact this, but I think the chance is very low that a product was deleted at the same time. Either way, checking that the text `1 product moved to the Trash.` is displayed on page is kind of the same, it doesn't guarantee that the message is not coming from other test.
If it's ok with you, I would leave it like this.

Reply:
Your point was good here and I can agree with you. I know that when I manually tested this page there are other messages shown with the same id `#message`. Also, I intended to bring one more time `getBy` here as well.

I believe we recently improved parallel execution and can be improved once again not to execute same file in parallel just to avoid this or other situations. Test file should be run sequentially if not already.

If we assume that the message will always be the last one, we can leave it as is. So, agreed. 👍